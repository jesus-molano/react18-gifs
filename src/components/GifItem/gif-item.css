.card {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: min(100%, 20rem);
  height: 20rem;
  background-color: #032b43;
  border-radius: 10px;
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
  transition: all 0.3s ease-in-out;
  overflow: hidden;

}
.card:hover {
  opacity: 0.8;
  transform: scale(1.05);
  box-shadow: 0px 0px 10px #0D8BBD;
}
.card img{
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 10px;
}
.card p{
  padding-top: 3rem;
  padding-inline:0.5rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  text-align: center;
  position: absolute;
  bottom: 0;
  background: rgb(2,0,36);
  background: linear-gradient(0deg, rgba(0,0,0,0.7) 50%, rgba(0,212,255,0) 100%);
  width: 100%;
  height: min(100%, 6rem);
  color: #fff;
}