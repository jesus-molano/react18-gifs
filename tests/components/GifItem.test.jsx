import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"


describe('Tests for GifItem component', () => {
  const title = 'Goku'
  const url = 'https://i.gifer.com/no.gif'

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('should render title and url', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should show title in the component', () => {
    render(<GifItem title={title} url={url} />)
    expect( screen.getByText(title)).toBeTruthy()
  })
})