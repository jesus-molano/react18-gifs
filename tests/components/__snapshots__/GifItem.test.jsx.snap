// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifItem component should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://i.gifer.com/no.gif"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
